lib/common/multithread.c

```diff
@@ -87,11 +87,16 @@ static void init_async(h2o_multithread_queue_t *queue, h2o_loop_t *loop)
 {
     int fds[2];
 
+#ifndef _WIN32
     if (cloexec_pipe(fds) != 0) {
         perror("pipe");
         abort();
     }
     fcntl(fds[1], F_SETFL, O_NONBLOCK);
+#else
+    u_long nonblock = 1;
```

Comment:
> In cloexec to setup fds[0] and fds[1] on windows we have to create the socket before hand
> This means we need to know if it is tcp or udp a stream etc beforehand which I don't think we know at this point ?
> This is done in the set_cloexec function

In the unix-side, fds[0] and fds[1] are initialized by calling `cloexec_pipe` (which in turn calls `pipe(2)`).  So on the Windows-side, the descriptors should be initialized as well, before setting them to non-blocking mode.

The file descriptors being created are used for signalling between the threads, and the threads listen to the sockets using the event loop.  So I think on Windows you would need to create a pair of TCP sockets connected to each other under my understanding that the HANDLEs created by `CreatePipe` cannot be passed to `select`.